### PR TITLE
Fix #2205 - make sure assembly resolution handler can't throw (V2)

### LINKED ISSUE #2205: Make sure our handler for AppDomain.AssemblyResolve cannot throw an exception
The specialization code path makes it possible for us to have two handlers registered on `AppDomain.CurrentDomain.AssemblyResolve`. The way this can happen is that we construct and assign a WebScriptHostManager for the main app, and then we call Stop() on the standby host manager. The main host manager might have hooked into AssemblyResolve before Stop() finished executing and unhooked the event. Because these actions are happening on different threads, the ordering is not deterministic.

The reason the above is important is because `AppDomain.CurrentDomain.AssemblyResolve` does not handle exceptions well. It will call the handlers one after another until one of them returns an Assembly or throws an exception (you can see this [here](https://referencesource.microsoft.com/#mscorlib/system/appdomain.cs,3157)). This means that if the AssemblyResolve handler for the standby host manages to throw an exception, it might prevent the AssemblyResolve handler for the user application from running. This can result in Assembly load failure errors.

In order to prevent the above we need to be extra defensive in our handler for AsssemblyResolve - even a logging call could throw an exception.

### PR DESCRIPTION
**Pull request by paulbatum**
Resolves https://github.com/Azure/azure-functions-host/issues/2205

### CODE REVIEW

File: src/WebJobs.Script/Description/DotNet/FunctionAssemblyLoader.cs [hunk 1]
```diff
@@ -47,7 +47,26 @@ private void Dispose(bool disposing)
             }
         }
 
+        /// <summary>
+        /// Note - it is possible for this method to be hooked up to AppDomain.CurrentDomain.AssemblyResolve multiple times, in different contexts (for example, during placeholder specialization).
+        /// The way AppDomain.CurrentDomain.AssemblyResolve works, if one registered handler throws an exception, the other handlers do not get to participate.
+        /// This behavior can result in assembly load failures that are difficult to diagnose or understand. To avoid this problem, this implementation swallows all exceptions.
+        /// Error logging, where possible, is performed in ResolveAssemblyCore.
+        /// </summary>
         internal Assembly ResolveAssembly(object sender, ResolveEventArgs args)
```
Comment (fabiocav) [inline src/WebJobs.Script/Description/DotNet/FunctionAssemblyLoader.cs]: FYI, all of this will be removed with my upcoming resolution/binding changes

**Comment by paulbatum** [comment]
@fabiocav maybe I should just abandon this change on V2? You're changing it anyway, I think it just creates slightly more work for you if I merge.

**Comment by fabiocav** [comment]
Yeah, I don't think it's worth checking those changes into V2 as I'm about to open a PR with the initial set of changes for load context and resolution.

**Comment by paulbatum** [comment]
Makes sense. Closing.

**Pull request closed without merging**